@PROTOBUF_SYNTAX_VERSION@
import "dccl/option_extensions.proto";
package dccl.test;

message ShortIDMsg
{
  option (dccl.msg).id = 2;
  option (dccl.msg).max_bytes = 1;
  option (dccl.msg).codec_version = 4;
}

message ShortIDMsgWithData
{
  option (dccl.msg).id = 3;
  option (dccl.msg).max_bytes = 10;
  option (dccl.msg).codec_version = 4;

  optional int32 in_head = 1 [(dccl.field).in_head=true, (dccl.field).min=0, (dccl.field).max=100];
  optional int32 in_body = 2 [(dccl.field).in_head=true, (dccl.field).min=0, (dccl.field).max=100];
}

message LongIDMsg
{
  option (dccl.msg).id = 10000;
  option (dccl.msg).max_bytes = 2;
  option (dccl.msg).codec_version = 4;
}

message TooLongIDMsg
{
  option (dccl.msg).id = 32768;
  option (dccl.msg).max_bytes = 32;
  option (dccl.msg).codec_version = 4;
}

message LongIDEdgeMsg
{
  option (dccl.msg).id = 128;
  option (dccl.msg).max_bytes = 2;
  option (dccl.msg).codec_version = 4;
}

message ShortIDEdgeMsg
{
  option (dccl.msg).id = 127;
  option (dccl.msg).max_bytes = 1;
  option (dccl.msg).codec_version = 4;
}
